### PR TITLE
[MillenniumDB] Add millenniumdb

### PR DESCRIPTION
**Pull request by jeremiahpslewis**
The following issues are pending prior to merge:

- https://github.com/MillenniumDB/MillenniumDB/issues/8
- https://github.com/MillenniumDB/MillenniumDB/issues/7
- https://github.com/MillenniumDB/MillenniumDB/issues/6
- https://github.com/MillenniumDB/MillenniumDB/issues/4
- https://github.com/MillenniumDB/MillenniumDB/issues/2

### CODE REVIEW

File: M/MillenniumDB/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+atomic_patch -p1 ${WORKSPACE}/srcdir/patches/remove-flags.patch
+
+cmake -H. -B$prefix -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN}
+cmake --build $prefix
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms(; experimental=true)
+filter!(p -> Sys.islinux(p) & (arch(p) == "x86_64"), platforms)
```
Comment (giordano) [inline M/MillenniumDB/build_tarballs.jl:26]: That's a rather small set of platforms

File: M/MillenniumDB/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+platforms = expand_cxxstring_abis(platforms)
+filter!(x -> cxxstring_abi(x) != "cxx03", platforms)
```
Comment (giordano) [inline M/MillenniumDB/build_tarballs.jl:28]: Why?

File: M/MillenniumDB/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    ExecutableProduct("server", :server),
+    ExecutableProduct("query", :query),
+    ExecutableProduct("create_db", :create_db), 
+    ExecutableProduct("check_bpts", :check_bpts), 
+    ExecutableProduct("check_extendible_hash", :check_extendible_hash), 
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+    Dependency("boost_jll"; compat="=1.71.0"),
```
Comment (giordano) [inline M/MillenniumDB/build_tarballs.jl:41]: Why 1.71? We have 1.76, which is also available on more platforms

Comment (jeremiahpslewis) [reply M/MillenniumDB/build_tarballs.jl:41]: Thanks (had copied it over from another recipe)

File: M/MillenniumDB/build_tarballs.jl (hunk 2 above)
Comment (jeremiahpslewis) [reply M/MillenniumDB/build_tarballs.jl:28]: I got errors with C++03 and it doesn't build on Mac yet

File: M/MillenniumDB/build_tarballs.jl (hunk 1 above)
Comment (jeremiahpslewis) [reply M/MillenniumDB/build_tarballs.jl:26]: Yep, it's not ideal, working on it. :)

File: M/MillenniumDB/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    GitSource("https://github.com/MillenniumDB/MillenniumDB.git", "b1df3251fa1e51ad160cae1e0a57553da93724be"),  # TODO: use main branch, this is a temporary fix
+    DirectorySource("./bundled"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/MillenniumDB
+
+if [[ "${target}" == x86_64-* ]] || [[ "${target}" == i686-* ]] || [[ "${target}" == armv7l-* ]] || [[ "${target}" == powerpc64le-* ]] || [[ "${target}" == armv6l-* ]]; then
+    powerpc64le
```
Comment (giordano) [inline M/MillenniumDB/build_tarballs.jl:19]:
```suggestion
```

File: M/MillenniumDB/build_tarballs.jl [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+sources = [
+    GitSource("https://github.com/MillenniumDB/MillenniumDB.git", "b1df3251fa1e51ad160cae1e0a57553da93724be"),  # TODO: use main branch, this is a temporary fix
+    DirectorySource("./bundled"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/MillenniumDB
+
+if [[ "${target}" == x86_64-* ]] || [[ "${target}" == i686-* ]] || [[ "${target}" == armv7l-* ]] || [[ "${target}" == powerpc64le-* ]] || [[ "${target}" == armv6l-* ]]; then
```
Comment (giordano) [inline M/MillenniumDB/build_tarballs.jl:18]:
```suggestion
if [[ "${target}" == x86_64-* ]] || [[ "${target}" == i686-* ]] || [[ "${target}" == arm-* ]] || [[ "${target}" == powerpc64le-* ]]; then
```
although I don't understand why don't you do
```sh
if [[ "${target}" == aarch64-* ]]; then
    # ...
else
    # ...
fi
```
:slightly_smiling_face:

File: M/MillenniumDB/build_tarballs.jl [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    DirectorySource("./bundled"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/MillenniumDB
+
+if [[ "${target}" == aarch64-* ]]; then
+    atomic_patch -p1 ${WORKSPACE}/srcdir/patches/remove-flags-aarch.patch
+else; then
```
Comment (giordano) [inline M/MillenniumDB/build_tarballs.jl:20]:
```suggestion
else
```

**Pull request closed without merging**